src/main/scala/nl/biopet/tools/bamstats/generate/Generate.scala

```diff
@@ -55,13 +55,26 @@ object Generate extends ToolCommand[Args] {
           getDictFromBam(cmdArgs.bamFile)
       }
 
-    val stats = extractStats(
-      bamFile = cmdArgs.bamFile,
-      bedFile = cmdArgs.bedFile,
-      sequenceDict = sequenceDict,
-      scatterMode = cmdArgs.scatterMode,
-      onlyUnmapped = cmdArgs.onlyUnmapped
-    )
+    val stats: GroupStats = cmdArgs.bedFile match {
+      case Some(bed: File) if cmdArgs.onlyUnmapped =>
+        throw new IllegalArgumentException(
+          "Cannot extract stats from regions and unmapped regions at the same time")
+      // When a BED file is specified extract regions stats
+      case Some(bed: File) if !cmdArgs.onlyUnmapped =>
+        val regions =
+          BedRecordList
+            .fromFile(bed)
+            .combineOverlap
+            .validateContigs(sequenceDict)
+        val regionStats = regions.allRecords.map { region =>
+          extractStatsRegion(cmdArgs.bamFile, region, cmdArgs.scatterMode)
+        }.toList
```

Comment:
Don't make this a list, this means all regions should be kept in memory.
When doing reduce directly on a Iterator there are at max 2 Stats objects in memory.

Because regionStats is not used for anything else this might also be an option:
```scala
          BedRecordList
            .fromFile(bed)
            .combineOverlap
            .validateContigs(sequenceDict)
            .allRecords.map { region =>
              extractStatsRegion(cmdArgs.bamFile, region, cmdArgs.scatterMode)
          }
          .reduce(_ += _)
```

Vals are also possible but make sure that are intermediate vals are Iterators and not Iterable (like seq, list, array)